src/Evolution/Systems/GeneralizedHarmonic/Equations.cpp

```diff
@@ -1,16 +1,44 @@
 // Distributed under the MIT License.
 // See LICENSE.txt for details.
-
 #include "Evolution/Systems/GeneralizedHarmonic/Equations.hpp"
 
 #include <array>
 
 #include "DataStructures/DataVector.hpp"
 #include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
 #include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "Evolution/Systems/GeneralizedHarmonic/Characteristics.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
 #include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+template <typename TagsList>
+class Variables;
 
 // IWYU pragma: no_forward_declare Tensor
+namespace {
+template <typename FieldTag>
+typename FieldTag::type weight_char_field(
+    const typename FieldTag::type& char_field_int,
+    const Scalar<DataVector>& char_speed_int,
+    const typename FieldTag::type& char_field_ext,
+    const Scalar<DataVector>& char_speed_ext) noexcept {
+  typename FieldTag::type weighted_char_field = char_field_int;
+  DataVector char_speed_avg = get(char_speed_int);
+  char_speed_avg -= 0.5 * (get(char_speed_int) - get(char_speed_ext));
+
+  auto weighted_char_field_it = weighted_char_field.begin();
```

Comment:
No good reason...just copied code from somewhere

Reply:
Turns out there was a reason...I don't know how to get the code to compile if I initialize them all inside the for loop, because the compiler has trouble deciphering what auto means when I try that. If someone else has a solution, I'm happy to make the change, but I'm not sure it's worth it